src/components/Home.js

```diff
@@ -1,11 +1,65 @@
-import React from 'react';
-
-const Home = ({ handleTokenCreation }) => (
-  <main>
-    <button onClick={handleTokenCreation} className="button-home">
-      Create a Shopping List
-    </button>
-  </main>
-);
+import React, { useState } from 'react';
+import { db } from '../lib/firebase';
+
+const Home = (props) => {
+  const [value, setValue] = useState('');
+  const [message, setMessage] = useState('');
+  const setToken = props.setToken;
+
+  const handleChange = (e) => setValue(e.target.value);
+
+  const joinList = (e) => {
+    e.preventDefault();
+    //stringify input value
+    const userToken = JSON.stringify(value);
```

Comment:
So this JSON.stringify is not needed. This is why we are getting errors when we try to put a token that's already in the firestore. When I pass a token like "sorb defy divan", it's going to work because in the collection the token has four quotes (""sorb defy divan""). However, If I try to put another token, it won't work. So removing that and updating the token in firestore to have only two quotes, it will work.

Reply:
Yeah, thanks for catching that! Fixing it now.